# Examining Racial Discrimination in the US Job Market

### Background
Racial discrimination continues to be pervasive in cultures throughout the world. Researchers examined the level of racial discrimination in the United States labor market by randomly assigning identical résumés to black-sounding or white-sounding names and observing the impact on requests for interviews from employers.

### Data
In the dataset provided, each row represents a resume. The 'race' column has two values, 'b' and 'w', indicating black-sounding and white-sounding. The column 'call' has two values, 1 and 0, indicating whether the resume received a call from employers or not.

Note that the 'b' and 'w' values in race are assigned randomly to the resumes when presented to the employer.

### Exercises
You will perform a statistical analysis to establish whether race has a significant impact on the rate of callbacks for resumes.

Answer the following questions **in this notebook below and submit to your Github account**. 

   1. What test is appropriate for this problem? Does CLT apply?
   2. What are the null and alternate hypotheses?
   3. Compute margin of error, confidence interval, and p-value. Try using both the bootstrapping and the frequentist statistical approaches.
   4. Write a story describing the statistical significance in the context or the original problem.
   5. Does your analysis mean that race/name is the most important factor in callback success? Why or why not? If not, how would you amend your analysis?

You can include written notes in notebook cells using Markdown: 
   - In the control panel at the top, choose Cell > Cell Type > Markdown
   - Markdown syntax: http://nestacms.com/docs/creating-content/markdown-cheat-sheet

#### Resources
+ Experiment information and data source: http://www.povertyactionlab.org/evaluation/discrimination-job-market-united-states
+ Scipy statistical methods: http://docs.scipy.org/doc/scipy/reference/stats.html 
+ Markdown syntax: http://nestacms.com/docs/creating-content/markdown-cheat-sheet
+ Formulas for the Bernoulli distribution: https://en.wikipedia.org/wiki/Bernoulli_distribution

In [2]:
import pandas as pd
import numpy as np
from scipy import stats

In [3]:
data = pd.io.stata.read_stata('data/us_job_market_discrimination.dta')

In [18]:
# number of callbacks for white-sounding names
white_callbacks = sum(data[data.race=='w'].call)
print white_callbacks

235.0


In [19]:
# number of callbacks for black-sounding names
black_callbacks = sum(data[data.race=='b'].call)
print black_callbacks

157.0


In [20]:
# number of no callbacks white-sounding names
white_nocallbacks = sum(data[data.race=='w'].call==0)
print white_nocallbacks

2200


In [21]:
# number of no callbacks for black-sounding names
black_nocallbacks = sum(data[data.race=='b'].call==0)
print black_nocallbacks

2278


In [6]:
data.head()

id ad  education  ofjobs  yearsexp  honors  volunteer  military  empholes  \
0  b  1          4       2         6       0          0         0         1   
1  b  1          3       3         6       0          1         1         0   
2  b  1          4       1         6       0          0         0         0   
3  b  1          3       4         6       0          1         0         1   
4  b  1          3       3        22       0          0         0         0   

   occupspecific    ...      compreq  orgreq  manuf  transcom  bankreal trade  \
0             17    ...          1.0     0.0    1.0       0.0       0.0   0.0   
1            316    ...          1.0     0.0    1.0       0.0       0.0   0.0   
2             19    ...          1.0     0.0    1.0       0.0       0.0   0.0   
3            313    ...          1.0     0.0    1.0       0.0       0.0   0.0   
4            313    ...          1.0     1.0    0.0       0.0       0.0   0.0   

  busservice othservice  missind  ownership  
0        0.0        0.0      0.0             
1        0.0        0.0      0.0             
2        0.0        0.0      0.0             
3        0.0        0.0      0.0             
4        0.0        1.0      0.0  Nonprofit  

[5 rows x 65 columns]

In [25]:
print(data.call.head(5))

0    0.0
1    0.0
2    0.0
3    0.0
4    0.0
Name: call, dtype: float32


<div class="span5 alert alert-success">
<p>Your answers to Q1 and Q2 here</p>
</div>
The Bernoulli distribution is the appropriate approach.
The Central Limit Theorem will apply if n > 30, which it is.

null hypothesis: H0: π1 = π2

alternative hypothesis: Ha : π1 < π2

where π1 and π2 are the sample proportions

In [44]:
w = data[data.race=='w']
b = data[data.race=='b']

In [52]:
white = white_callbacks + white_nocallbacks
black = black_callbacks + black_nocallbacks
pi1 = white_callbacks / (white)
pi2 = black_callbacks / (black)
print("pi1 is " + str(pi1) + ", and pi2 is: " + str(pi2))
pihat = (white_callbacks + black_callbacks) / (white + black)
print("pihat is: " + str(pihat))
z = (pi1 + pi2) / np.sqrt(pihat*(1 - pihat) * (1/(white))+1/(black)) # formula from https://www.cliffsnotes.com/study-guides/statistics/univariate-inferential-tests/test-for-comparing-two-proportions
print("The z value is: " + str(z))

pi1 is 0.0965092402464, and pi2 is: 0.064476386037
pihat is: 0.0804928131417
The z value is: 7.665344352399563


We need to find the z score for alpha value 0.1.  For a one-sided distribution, subtract 0.1 from 1 to get 0.9.  Look for 0.9 in the table.  The standard normal ( z) table shows that the lower critical z‐value for α = 0.10 is approximately –1.28. The computed z must be lower than –1.28 in order to reject the null hypothesis of equal proportions. Because the computed z is 7.67, the null hypothesis can be rejected. It can be concluded (at this level of significance) that whites receive significantly more callbacks than blacks.

Upper confidence interval:

In [59]:
upperCI = pi1 - pi2 + (1 - .1)*np.sqrt(pi1*(1-pi1)/white + pi2*(1-pi2)/black)
print upperCI

0.03903788773745467


Lower confidence interval:

In [60]:
lowerCI = pi1 - pi2 - (1 - .1)*np.sqrt(pi1*(1-pi1)/white + pi2*(1-pi2)/black)
print lowerCI

0.025027820681436507


"Recall that the margin of error is the distance between the sample proportion M and an endpoint of the confidence interval."
http://www.randomservices.org/random/interval/Bernoulli.html

In [62]:
margin_of_error = (1 - .1)*np.sqrt(pi1*(1-pi1)/white + pi2*(1-pi2)/black)
print margin_of_error

0.007005033528009079


<div class="span5 alert alert-success">
<p> Your answers to Q4 and Q5 here </p>
</div>

We sought to verify whether race plays a significant role in callback from job applications.  Statistically speaking, there is a role of race in the percentage of callbacks as indicated by name.

This analysis does not indicate that race is the most important factor in receiving a callback.  There are a number of other factors that could play a larger role, and I would calculate their impact on callback probability.